app/code/Magento/Ui/view/base/web/js/form/element/date.js

```diff
@@ -177,9 +177,9 @@ define([
          * with moment.js library.
          */
         prepareDateTimeFormats: function () {
-            this.pickerDateTimeFormat = this.options.dateFormat;
+            this.pickerDateTimeFormat = this.options.timeOnly ? this.options.timeFormat : this.options.dateFormat;
```

Comment:
Could you rewrite L180-L184 to make it more readable
```
if (this.options.timeOnly) {
    this.pickerDateTimeFormat = this.options.timeFormat;
} else {
    this.pickerDateTimeFormat = this.options.dateFormat;
    if (this.options.showsTime) {
        this.pickerDateTimeFormat += ' ' + this.options.timeFormat;
    }
}
```

Reply:
Fixed.